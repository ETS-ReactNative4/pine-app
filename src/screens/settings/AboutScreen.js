import React, { Component } from 'react';
import { Linking } from 'react-native';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import headerStyles from '../../styles/headerStyles';
import BackButton from '../../components/BackButton';
import SettingsGroup from '../../components/SettingsGroup';
import SettingsAttribute from '../../components/SettingsAttribute';
import SettingsButton from '../../components/SettingsButton';
import SettingsLink from '../../components/SettingsLink';
import BaseSettingsScreen from './BaseSettingsScreen';
import config from '../../config';

const packageJson = require('../../../package.json');

@connect()
export default class AboutScreen extends Component {
  static navigationOptions = ({ navigation }) => ({
    title: 'About',
    headerStyle: headerStyles.header,
    headerTitleStyle: headerStyles.title,
    headerLeft: (<BackButton onPress={() => { navigation.goBack(); }} />)
  });

  _showTermsAndConditions() {
    const navigation = this.props.navigation;
    navigation.navigate('TermsAndConditions');
  }

  _visitWebsite() {
    Linking.openURL(packageJson.website);
  }

  _reportIssue() {
    Linking.openURL(packageJson.bugs);
  }

  _viewSourceCode() {
    Linking.openURL(packageJson.repository.url);
  }

  render() {
    return (
      <BaseSettingsScreen>
        <SettingsGroup>
          <SettingsAttribute name='Version' value={packageJson.version} />
          <SettingsAttribute name='Copyright' value={packageJson.copyright} isLastItem={true} />
        </SettingsGroup>

        <SettingsGroup>
          <SettingsAttribute name='Bitcoin Network' value={config.bitcoin.network} isLastItem={true} />
        </SettingsGroup>

        <SettingsGroup>
          <SettingsLink name='Terms and Conditions' onPress={this._showTermsAndConditions.bind(this)} isLastItem={true} />
        </SettingsGroup>

        <SettingsGroup>
          <SettingsButton title='Visit Website' onPress={this._visitWebsite.bind(this)} />
          <SettingsButton title='Report Issue' onPress={this._reportIssue.bind(this)} />
          <SettingsButton title='View Source Code' onPress={this._viewSourceCode.bind(this)} isLastItem={true} />
        </SettingsGroup>
      </BaseSettingsScreen>
    );
  }
}

AboutScreen.propTypes = {
  dispatch: PropTypes.func,
  navigation: PropTypes.any
};
