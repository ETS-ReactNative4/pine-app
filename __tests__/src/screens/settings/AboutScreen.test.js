import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import AboutScreen from '../../../../src/screens/settings/AboutScreen';

jest.mock('../../../../src/containers/ErrorModalContainer', () => 'ErrorModalContainer');

const storeMock = {
  getState: jest.fn(() => ({})),
  dispatch: jest.fn()
};

const navigationMock = {
  navigate: jest.fn(),
  state: {
    params: {}
  }
};

describe('AboutScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <AboutScreen store={storeMock} navigation={navigationMock} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
