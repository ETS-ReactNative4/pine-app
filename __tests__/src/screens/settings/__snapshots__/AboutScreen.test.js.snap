// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#EFEFF3",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "paddingTop": 35,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#C8C8CC",
                "borderBottomWidth": 0.5,
                "borderTopColor": "#C8C8CC",
                "borderTopWidth": 0.5,
                "marginBottom": 35,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 0.5,
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                  "marginLeft": 15,
                  "paddingRight": 15,
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "fontSize": 17,
                    "fontWeight": "400",
                    "lineHeight": 22,
                  },
                ]
              }
            >
              Version
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#8F8E94",
                    "fontSize": 17,
                    "fontWeight": "400",
                    "lineHeight": 22,
                    "position": "absolute",
                    "right": 15,
                    "textAlign": "right",
                    "width": 250,
                  },
                ]
              }
            >
              0.0.1
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 0.5,
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                  "marginLeft": 15,
                  "paddingRight": 15,
                },
                Object {
                  "borderBottomWidth": 0,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "fontSize": 17,
                    "fontWeight": "400",
                    "lineHeight": 22,
                  },
                ]
              }
            >
              Copyright
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#8F8E94",
                    "fontSize": 17,
                    "fontWeight": "400",
                    "lineHeight": 22,
                    "position": "absolute",
                    "right": 15,
                    "textAlign": "right",
                    "width": 250,
                  },
                ]
              }
            >
              Â© 2018 Blockfirm AB
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#C8C8CC",
                "borderBottomWidth": 0.5,
                "borderTopColor": "#C8C8CC",
                "borderTopWidth": 0.5,
                "marginBottom": 35,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 0.5,
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                  "marginLeft": 15,
                  "paddingRight": 15,
                },
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "fontSize": 17,
                    "fontWeight": "400",
                    "lineHeight": 22,
                  },
                ]
              }
            >
              Bitcoin Network
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#8F8E94",
                    "fontSize": 17,
                    "fontWeight": "400",
                    "lineHeight": 22,
                    "position": "absolute",
                    "right": 15,
                    "textAlign": "right",
                    "width": 250,
                  },
                ]
              }
            >
              testnet
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "borderBottomColor": "#C8C7CC",
                  "borderBottomWidth": 0.5,
                  "flex": 1,
                  "height": 44,
                  "justifyContent": "center",
                  "marginLeft": 15,
                  "paddingRight": 15,
                },
                Object {
                  "borderBottomWidth": 0,
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={undefined}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "fontSize": 17,
                    "fontWeight": "400",
                    "lineHeight": 22,
                  },
                ]
              }
            >
              Key Derivation Method
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#8F8E94",
                    "fontSize": 17,
                    "fontWeight": "400",
                    "lineHeight": 22,
                    "position": "absolute",
                    "right": 15,
                    "textAlign": "right",
                    "width": 250,
                  },
                ]
              }
            >
              BIP44
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#C8C8CC",
                "borderBottomWidth": 0.5,
                "borderTopColor": "#C8C8CC",
                "borderTopWidth": 0.5,
                "marginBottom": 35,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#C8C7CC",
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                    "height": 44,
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "paddingRight": 15,
                  },
                  Object {
                    "borderBottomWidth": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Object {
                      "fontSize": 17,
                      "fontWeight": "400",
                      "lineHeight": 22,
                    },
                  ]
                }
              >
                Terms and Conditions
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#8F8E94",
                        "fontSize": 17,
                        "fontWeight": "400",
                        "lineHeight": 22,
                        "position": "absolute",
                        "right": 15,
                        "textAlign": "right",
                        "width": 250,
                      },
                      Object {
                        "right": 35,
                      },
                    ],
                  ]
                }
              />
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": undefined,
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#C7C7CC",
                      "fontSize": 20,
                      "paddingTop": 2,
                      "position": "absolute",
                      "right": 15,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                ï
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderBottomColor": "#C8C8CC",
                "borderBottomWidth": 0.5,
                "borderTopColor": "#C8C8CC",
                "borderTopWidth": 0.5,
                "marginBottom": 35,
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#C8C7CC",
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                    "height": 44,
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "fontSize": 17,
                        "fontWeight": "400",
                        "lineHeight": 22,
                      },
                      Object {
                        "color": "#007AFF",
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Visit Website
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#C8C7CC",
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                    "height": 44,
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "paddingRight": 15,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "fontSize": 17,
                        "fontWeight": "400",
                        "lineHeight": 22,
                      },
                      Object {
                        "color": "#007AFF",
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                Report Issue
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={null}
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Array [
                  Object {
                    "borderBottomColor": "#C8C7CC",
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                    "height": 44,
                    "justifyContent": "center",
                    "marginLeft": 15,
                    "paddingRight": 15,
                  },
                  Object {
                    "borderBottomWidth": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={undefined}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    Array [
                      Object {
                        "fontSize": 17,
                        "fontWeight": "400",
                        "lineHeight": 22,
                      },
                      Object {
                        "color": "#007AFF",
                      },
                      undefined,
                      undefined,
                    ],
                  ]
                }
              >
                View Source Code
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <ErrorModalContainer />
</View>
`;
